lib/chef/provisioning/transport/winrm.rb

```diff
@@ -86,14 +86,14 @@ def available?
         # If you can't pwd within 10 seconds, you can't pwd
         execute('pwd', :timeout => 10)
         true
-      rescue Timeout::Error, Errno::EHOSTUNREACH, Errno::ETIMEDOUT, Errno::ECONNREFUSED, Errno::ECONNRESET, ::WinRM::WinRMHTTPTransportError, ::WinRM::WinRMWebServiceError, ::WinRM::WinRMWSManFault
-        Chef::Log.debug("unavailable: network connection failed or broke: #{$!.inspect}")
-        disconnect
-        false
       rescue ::WinRM::WinRMAuthorizationError
         Chef::Log.debug("unavailable: winrm authentication error: #{$!.inspect} ")
         disconnect
         false
+      rescue network_errors
```

Comment:
You can "splat" these so you don't have to make a container module below. For example:

```ruby
class A < StandardError; end
class B < StandardError; end
class C < StandardError; end
errors = [ A, B ]

begin
  raise A, "a"
rescue *errors => e
  puts e
end
# => a
begin
  raise B, "b"
rescue *errors => e
  puts e
end
# => b
begin
  raise C, "c"
rescue *errors => e
  puts e
end
# => does not rescue
```